adoc/chapters/programming_interface.adoc

```diff
@@ -4976,20 +4989,20 @@ with attached host memory and is still in use.
 
 More precisely:
 
-  . A buffer can be constructed with just a size and using the default
-    buffer allocator. The memory management for this type of buffer is
-    entirely handled by the SYCL system. The destructor for this type of
+  . A buffer can be constructed from a [code]#range# (and without a
+    [code]#hostData# pointer).  The memory management for this type of buffer
+    is entirely handled by the SYCL system. The destructor for this type of
     buffer does not need to block, even if work on the buffer has not
     completed. Instead, the SYCL system frees any storage required for the
     buffer asynchronously when it is no longer in use in queues. The initial
     contents of the buffer are unspecified.
-  . A buffer can be constructed with associated host memory and a default
-    buffer allocator. The buffer will use this host memory for its full
-    lifetime, but the contents of this host memory are unspecified for the
-    lifetime of the buffer. If the host memory is modified by the host, or
-    mapped to another buffer or image during the lifetime of this buffer,
-    then the results are undefined. The initial contents of the buffer will
-    be the contents of the host memory at the time of construction.
+  . A buffer can be constructed from a [code]#hostData# pointer.  The buffer
+    will use this host memory for its full lifetime, but the contents of this
+    host memory are unspecified for the lifetime of the buffer. If the host
+    memory is modified by the host or if it is used to construct another
```

Comment:
```suggestion
    memory is modified on the host or if it is used to construct another
```
to avoid weird case where an RDMA from the outer-space is changing the value.